tofhir-server/src/main/scala/io/tofhir/server/service/SchemaDefinitionService.scala

```diff
@@ -100,8 +102,71 @@ class SchemaDefinitionService(schemaRepository: ISchemaRepository, mappingReposi
    */
   def inferSchema(inferTask: InferTask): Future[Option[SchemaDefinition]] = {
     // Execute SQL and get the dataFrame
-    val dataFrame = SourceHandler.readSource(inferTask.name, SparkConfig.sparkSession,
-      inferTask.sourceContext, inferTask.sourceSettings.head._2, None, None, Some(1))
+    val dataFrame = try {
+      SourceHandler.readSource(inferTask.name, SparkConfig.sparkSession,
+        inferTask.sourceContext, inferTask.sourceSettings.head._2, None, None, Some(1))
+    } catch {
```

Comment:
Here, only keep the exception logic that is directly related to the inferral business logic. Move the source reading-related logic into SourceHandler.

Reply:
I did it. But ToFhirError class is in server package and SourceHandler is in engine package. I could not import ToFhirError in it. So I could not handle errors with that much detail.